vllm/v1/attention/backends/flash_attn.py

```diff
@@ -428,6 +470,14 @@ def forward(
             # Profiling run.
             return output
 
+        # Validate attention metadata based on attention type
+        attn_type = self.attn_type
+        if (attn_type in (AttentionType.ENCODER, AttentionType.ENCODER_DECODER,
+                          AttentionType.ENCODER_ONLY)
+                and (not attn_metadata.is_all_encoder_attn_metadata_set)):
+            raise AttributeError("Encoder attention requires setting "
+                                 "encoder metadata attributes.")
```

Comment:
ENCODER_ONLY attention doesn't need this, can use the same attributes as DECODER.